### PR TITLE
Add Additional Frameworks and Testing Terms

### PR DESCRIPTION
**Pull request by cprime**

### CODE REVIEW

File: ios/ios_testing_references.md [hunk 1]
```diff
@@ -18,47 +18,55 @@ UI testing is the process of ensuring proper functionality of the UI for a given
...
+The iOS team is currently focused on writing unit tests against Models and ViewModels. By adopting an MVVM architecture, most business logic will be extracted from the ViewController and included in a ViewModel, thereby making this business logic more testable. For a more detailed look at Testing and MVVM see this related [reference] (https://github.com/IntrepidPursuits/sherpa/blob/master/ios/MVVM_Testing.md).
 
 ## Testing Tools
 
-Provided below is a list of commonly used tools when writing tests. 
+Provided below is a list of commonly used tools when writing tests.
+
+### OHHTTPStubs - Web stubbing
+
+Stubbing webcalls is required when unit testing portions of a codebase that relied on web resources. The desired stubbing tool is [OHHTTPStubs](https://github.com/AliSoftware/OHHTTPStubs). OHHTTPStubs allow for stubbing of specific webcalls or all web traffic. Stubbed webcalls are intercepted and allow for a static file (JSON/XML/etc) to be returned.
```
Comment (vinnyt) [inline ios/ios_testing_references.md]: relied -> rely
I know there is a knowledge gap here having watching some people try to write tests against web endpoints and just calling the real thing and not even the slightest idea that might be a good idea. I think we need to provide explanation as to why.

Comment (vinnyt) [reply ios/ios_testing_references.md]: Also should expand they will need to interact with the backend team to get fixture that represent the results of calls.

File: ios/ios_testing_references.md [hunk 2]
```diff
@@ -68,9 +76,50 @@ class SearchViewModel {
...
+    let mockNetworkingLayer = MockNetworkingLayer()
+    let viewModel = SearchViewModel(networkingLayer: mockNetworkingLayer)
+    mockNetworkingLayer.stubbedResult = Result.failure(APIError())
+    viewModel.searchFor(searchTerm: "Big")
+    // verify SearchViewModel properly handles failed searches
+    ...
+}
+
+```
+
```
Comment (vinnyt) [inline ios/ios_testing_references.md:138]: should probably also expand this to include spying which is essentially using a mock but setting it up such that you can verify a particular consumer of a class is calling the methods they are expected to call.

File: ios/ios_testing_references.md [hunk 3]
```diff
@@ -68,9 +76,50 @@ class SearchViewModel {
 
 In the above example, the `SearchViewModel` can be initialized with a `NetworkingLayer` to allow a mock `NetworkingLayer` to be injected into the `SearchViewModel`. In this example `NetworkingLayer` is a protocol and as a result the live and mock network layers can both conform to the protocol. The class `DefaultNetworkingLayer` conforms to `NetworkingLayer` and is included as a default initializer. Using protocol is an effective way to implement dependency injection.
 
+### Mocking
+
+As eluded to in the previous section, mocking is a process of creating a dummy or 'mock' object, which will simulate the behavior of a real dependency. This is useful if the real objects are impractical to incorporate into a unit test. Objective-C has access to tools like OCMockito which allow you to automatically generate mock objects at runtime. Unfortunately, because of limitations in the Swift language, you are forced to write your own mocks.
```
Comment (vinnyt) [inline ios/ios_testing_references.md]: I would leave out this last part about Mockito.  IMHO if you are testing using protocols properly and testing via an interface it's actually an advantage of swift that you build your own, and are not dragging around a massive thing like mockito

**Review by vinnyt — COMMENTED** [review]
just a few tweaks.

File: ios/ios_testing_references.md (hunk 2 above)
Comment (cprime) [reply ios/ios_testing_references.md:138]: Agreed, I will add a section on this.

File: ios/ios_testing_references.md (hunk 3 above)
Comment (cprime) [reply ios/ios_testing_references.md]: 👍

File: ios/ios_testing_references.md (hunk 1 above)
Comment (cprime) [reply ios/ios_testing_references.md]: 👍

File: ios/ios_testing_references.md (hunk 2 above)
Comment (cprime) [reply ios/ios_testing_references.md:138]: added

**Review by vinnyt — APPROVED** [review]
LGTM

File: ios/ios_testing_references.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
 # iOS Testing References
 
+## Why do we write tests?
+
+Project stakeholders are not always sold on the benefits of writing tests. Here are a few arguments to help convince them:
+
+* Automated tests allow you to catch bugs as they are being introduced, instead of being caught by users
+* You can integrate new code faster and with more confidence if there are tests looking for regression bugs
```
Comment (vinnyt) [inline ios/ios_testing_references.md]: I would rephrase this around the idea that it will catch regressions. This might confuse people thinking they need to write specific tests.
